daemons/lvmdbusd/utils.py

```diff
@@ -577,13 +578,21 @@ def done(self):
 		with self.cond:
 			if not self.function_complete:
 				self.cond.wait()
+		if self.exception:
+			raise self.exception
 		return self.rc
 
 	def _run(self):
-		if len(self.args):
-			self.rc = self.f(*self.args)
-		else:
-			self.rc = self.f()
+		try:
+			if len(self.args):
```

Comment:
I know you are not changing this line, but it should be ``if self.args``. That way ``self.args`` could also be ``None`` and it would still work. Plus ``if len()`` without a comparison is a bit confusing.

Reply:
Done, thanks!